integration-test/1810-alternate-viewpoints.py

```diff
@@ -0,0 +1,153 @@
+                    'source': 'openstreetmap.org',
+                    'mz_boundary_from_polygon': True,  # need this for hack
+                }),
+            dsl.way(4, dsl.fit_in_tile(
+                z, x, y,
+                'LINESTRING(0.1 0.5, 0.1 0.1, 0.9 0.1, 0.9 0.9, 0.5 0.9)'), {
+                    'admin_level': '2',
+                    'boundary': 'administrative',
+                    'name': 'YY',
+                    'source': 'openstreetmap.org',
+                    'mz_boundary_from_polygon': True,  # need this for hack
+                }),
+            # this is just here to turn off maritime boundaries for everything
+            # in this tile.
+            dsl.way(5, dsl.tile_box(z, x, y), {
+                'source': 'tilezen.org',
+                'maritime_boundary': True,
+                'min_zoom': 0,
+                'kind': 'maritime',
+            }),
+        )
+
+        # should get a non-disputed section of border
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'country',
+                'name:left': 'XX',
+                'name:right': 'YY',
+            })
+
+        # should get a section recognised only by XX
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'unrecognized_country',
+                'kind:xx': 'country',
+            })
+
+        # should get a section recognised _except_ by XX
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'country',
+                'kind:xx': 'unrecognized_country',
+            })
+
+    def test_claim(self):
+        # test that a claim by countries BB & CC (and recognised by DD) is
+        # only kind:country for those countries. everyone else's view is
+        # kind: unrecognized_country.
+        #
+        # TODO: recognized_by not working yet.
+        import dsl
+
+        z, x, y = (8, 0, 0)
+
+        self.generate_fixtures(
+            dsl.way(1, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '2',
+                'boundary': 'claim',
+                'name': 'Extent of CC claim',
+                'claimed_by': 'CC',
+                'disputed_by': 'AA;DD',
+            }),
+            dsl.way(2, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '2',
+                'boundary': 'claim',
+                'name': 'Extent of BB claim',
+                'claimed_by': 'BB',
+                'disputed_by': 'AA',
+                'recognized_by': 'DD',
+            }),
+            dsl.way(3, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '3',
+                'boundary': 'administrative',
+                'name': 'Region Name',
+                'type': 'boundary',
+            }),
+            dsl.way(4, dsl.tile_diagonal(z, x, y), {
+                'dispute': 'yes',
+                'disputed_by': 'AA',
+                'name': 'BB claim',  # note: also CC claim?!
```

Comment:
Do we test for this and the later ways here?

Reply:
I think it's already a case covered by the previous test but, just to be sure, I've added a further assertion in 68cbdbf1.